src/Multiplier.php

```diff
@@ -168,8 +169,10 @@ public function addCreateButton(?string $caption = null, int $copyCount = 1): Cr
 	 */
 	public function validate(?array $controls = null): void
 	{
-		/** @var Control[] $components */
-		$components = $controls ?? iterator_to_array($this->getComponents());
+		/** @var iterable<Control> $components */
```

Comment:
I suggest to extract logic into separate function/method. What do you think?

Reply:
Did that, thanks.